src/FishjamClient.ts

```diff
@@ -387,9 +397,9 @@ export class FishjamClient<
     this.initConnection(config.peerMetadata);
   }
 
-  private initConnection(peerMetadata: PeerMetadata): void {
+  private async initConnection(peerMetadata: PeerMetadata): Promise<void> {
     if (this.status === 'initialized') {
-      this.disconnect();
+      await this.disconnect();
```

Comment:
Maybe it would be worth to investigate if [this rule](https://typescript-eslint.io/rules/no-floating-promises/) can be used to detect issues like this one

Reply:
In general, I absolutely agree with you; we should add this rule. What's more, we should make more functions async, like connect, but I don't want to introduce such changes in this PR.

In this particular line, it's a relic of my previous attempts. The `disconnect()` function is not async.